### PR TITLE
Add GetDetailedExceptionMessage extension

### PR DESCRIPTION
**Pull request by LucGenetier**
Add more detailed information when LSP captures an exception

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Core.Tests/CoreUtilExtensionTests.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        {
+            try
+            {
+                var x = 0;
+                var y = 1 / x;
+            }
+            catch (Exception ex)
+            {
+                var result = ex.GetDetailedExceptionMessage();
+                Assert.Contains(@"Exception System.DivideByZeroException: Message='Attempted to divide by zero.', HResult=0x80020012, StackTrace='   at Microsoft.PowerFx.Core.Tests.CoreUtilExtensionTests.GetDetailedExceptionMessage_DivisionByZero() in C:\Data\2\Power-Fx\src\tests\Microsoft.PowerFx.Core.Tests\CoreUtilExtensionTests.cs", result);
```
[the remark this answers is not in the document]
Comment (LucGenetier) [reply src/tests/Microsoft.PowerFx.Core.Tests/CoreUtilExtensionTests.cs]: Nope !

File: src/libraries/Microsoft.PowerFx.Core/Utils/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        private static string GetDetailedExceptionMessageInternal(string prefix, Exception ex, int depth, int maxDepth)
+        {
+            depth++;
+
+            if (depth > maxDepth)
+            {
+                return string.Empty;
+            }
+
+            var sb = new StringBuilder();
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Utils/Extensions.cs:28]: > var sb = new StringBuilder(); [](http://example.com/codeflow?start=11&length=30)

nit - A pattern here is to create the stringBuilder once on line 15 and pass it in.  That will play better with recursion. 

See  FormulaValue.ToExpression as an example.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**